### PR TITLE
Usability and visual design improvements for devmode gizmo and notifications

### PR DESCRIPTION
**Pull request by jouni**
Partial redesign of the UI design. Wanted to clean up the visual hierarchy and improve the usability of the notifications (dismissing and the “don’t show again” feature).

As a disclaimer: I didn’t pay any attention to accessibility, as it looked like that wasn’t a concern previously either (color contrast, keyboard or screen reader usability). The gizmo is only usable with a mouse or a touch device, and by sighted people.

Apologies for the unnecessary white-space changes (my IDE does the formatting automatically). I didn’t bother leaving those out.

Not super happy about the code quality, but it does the job.

## Changes

- Refined typographic styles (slightly smaller overall)
- Visually detached the gizmo and notifications from the edge of the browser
- Added a blue color for “information” messages
- Added a shadow for the gizmo and notifications
- Added an explicit “Dismiss” button (previously the user could click anywhere on the notification to dismiss it)

Frame 1

- Moved the “Read more” link to be part of the message details
- Changed the “Don’t show again” from a button to a checkbox
- Added a “log” style message, which does not have an icon and uses the gray color

Frame 2

- Simplified the window layout: removed the title bar (moved the title to a message in the log)
- Redesigned minimize icon

Frame 3

- If there’s a notification at the same time as the window is open, the notification shows above the window (previously was on top of it)

Frame 4

- Allow the log window to scroll if there are many message (previously the window grew outside the browser viewport and it was impossible to minimize the window)
- In general, restricted the UI from growing outside the browser viewport (long messages, small viewports, etc.)

Frame 5

### CODE REVIEW

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 1]
```diff
@@ -787,69 +956,59 @@ class VaadinDevmodeGizmo extends LitElement {
...
+          ${messageObject.persistentId ? html`<div class="persist ${messageObject.dontShowAgain ? 'on' : 'off'}" @click=${e => this.toggleDontShowAgain(messageObject.id)}>Don’t show again</div>` : ''}
         </div>
+        <div class="dismiss-message" @click=${e => this.dismissNotification(messageObject.id)}>Dismiss</div>
       </div>
       `;
   }
 
   render() {
     return html`
-      <div class="vaadin-devmode-gizmo-root">
```
Comment (jouni) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:803]: Removed the unnecessary wrapper element, as it only seemed to serve the purpose of one CSS selector.

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 2]
```diff
@@ -681,17 +830,21 @@ class VaadinDevmodeGizmo extends LitElement {
 
   showSplashMessage(msg) {
     this.splashMessage = msg;
-    // automatically move notification to message tray after a certain amount of time
     if (this.splashMessage != null) {
-      setTimeout(() => {
+      if (this.expanded) {
         this.demoteSplashMessage();
```
Comment (jouni) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:835]: Show splash messages immediately inside the window if it is open.

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 3]
```diff
@@ -657,6 +804,8 @@ class VaadinDevmodeGizmo extends LitElement {
       this.showSplashMessage('Automatic reload #' + count + ' finished at ' + reloaded);
       window.sessionStorage.removeItem(VaadinDevmodeGizmo.TRIGGERED_KEY_IN_SESSION_STORAGE);
     }
+
+    this.__transitionDuration = parseInt(window.getComputedStyle(this).getPropertyValue('--gizmo-transition-duration'));
```
Comment (jouni) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:808]: Removed hardcoded timeout durations and used the single value from CSS.

**Comment by joheriks** [comment]
The linter complains about the two non-breaking space characters (0xA0) at line 959 col 121, and line 962, col 76.

**Comment by jouni** [comment]
> The linter complains about the two non-breaking space characters (0xA0) at line 959 col 121, and line 962, col 76.

Thanks! Fixed.

**Pull request merged**